## Week-1 solution

In [1]:
import pandas as pd

In [11]:
dfA = pd.read_csv("C:\\Users\\khurs\\OneDrive\\Desktop\\ML Class Code\\drb-sp26-ml\\week-01\\datasets\\A.csv")
dfB = pd.read_csv("C:\\Users\\khurs\\OneDrive\\Desktop\\ML Class Code\\drb-sp26-ml\\week-01\\datasets\\B.txt")


In [15]:
print(dfA.info())


<class 'pandas.DataFrame'>
RangeIndex: 9000 entries, 0 to 8999
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   CustomerId       9000 non-null   int64  
 1   Surname          9000 non-null   str    
 2   CreditScore      9000 non-null   int64  
 3   Geography        9000 non-null   str    
 4   Gender           9000 non-null   str    
 5   Age              9000 non-null   int64  
 6   Tenure           9000 non-null   int64  
 7   Balance          9000 non-null   float64
 8   NumOfProducts    9000 non-null   int64  
 9   HasCrCard        9000 non-null   int64  
 10  IsActiveMember   9000 non-null   int64  
 11  EstimatedSalary  9000 non-null   float64
 12  Exited           9000 non-null   int64  
dtypes: float64(2), int64(8), str(3)
memory usage: 914.2 KB
None


In [16]:
print(dfB.info())

<class 'pandas.DataFrame'>
RangeIndex: 232042 entries, 0 to 232041
Data columns (total 1 columns):
 #   Column                                                                                                Non-Null Count   Dtype
---  ------                                                                                                --------------   -----
 0   AQC00914000 1981  4279   3745  10762   6067   4096   3606   6203   5292   3092   6866   7163   7866   232042 non-null  str  
dtypes: str(1)
memory usage: 1.8 MB
None


In [19]:
dfC = pd.read_csv(
    "C:\\Users\\khurs\\OneDrive\\Desktop\\ML Class Code\\drb-sp26-ml\\week-01\\datasets\\C.csv", 
    sep=';', 
    skiprows=[1]
)

In [21]:
dfC.info()

<class 'pandas.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 16 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   name      77 non-null     str    
 1   mfr       77 non-null     str    
 2   type      77 non-null     str    
 3   calories  77 non-null     int64  
 4   protein   77 non-null     int64  
 5   fat       77 non-null     int64  
 6   sodium    77 non-null     int64  
 7   fiber     77 non-null     float64
 8   carbo     77 non-null     float64
 9   sugars    77 non-null     int64  
 10  potass    77 non-null     int64  
 11  vitamins  77 non-null     int64  
 12  shelf     77 non-null     int64  
 13  weight    77 non-null     float64
 14  cups      77 non-null     float64
 15  rating    77 non-null     float64
dtypes: float64(5), int64(8), str(3)
memory usage: 9.8 KB


How many rows are in dataset: `week-01/datasets/A.csv`? How about in `week-01/datasets/B.txt` and in
`week-01/datasets/C.csv`? 

A -> 9000
B -> 232042 
C -> 77

Repeat the previous task to find out how many `samples` (i.e., instances) are there in each of the dataset?

In [23]:
print(f"Number of samples (years): {len(dfA)}")
print(f"Number of samples (years): {len(dfB)}")
print(f"Number of samples (years): {len(dfC)}")


Number of samples (years): 9000
Number of samples (years): 232042
Number of samples (years): 77


How many columns are there in each of the 3 datasets?

In [24]:
print(dfA.columns)
print(dfB.columns)
print(dfC.columns)


Index(['CustomerId', 'Surname', 'CreditScore', 'Geography', 'Gender', 'Age',
       'Tenure', 'Balance', 'NumOfProducts', 'HasCrCard', 'IsActiveMember',
       'EstimatedSalary', 'Exited'],
      dtype='str')
Index(['AQC00914000 1981  4279   3745  10762   6067   4096   3606   6203   5292   3092   6866   7163   7866 '], dtype='str')
Index(['name', 'mfr', 'type', 'calories', 'protein', 'fat', 'sodium', 'fiber',
       'carbo', 'sugars', 'potass', 'vitamins', 'shelf', 'weight', 'cups',
       'rating'],
      dtype='str')


6. Compute the mean of the last (i.e., rightmost) column of `week-01/datasets/C.csv`?

In [25]:
last_column = dfC.iloc[:, -1]

# 3. Compute the mean
c_mean = last_column.mean()

print(f"Mean of the last column ({last_column.name}): {c_mean}")

Mean of the last column (rating): 42.66570498701299


In [26]:
dfD = pd.read_csv("C:\\Users\\khurs\\OneDrive\\Desktop\\ML Class Code\\drb-sp26-ml\\week-01\\datasets\\D.txt")

In [27]:
differences = (dfB != dfD).any(axis=1)

# 3. Get the indices (row numbers) where they differ
diff_indices = dfB.index[differences].tolist()

print(f"The datasets differ at sample indices: {diff_indices}")
# If you want to see the actual differing rows:
# print(dfB.loc[diff_indices])

The datasets differ at sample indices: [112588]


In [28]:
# 2. Inspect column names to find the exact spelling
print("Columns in A:", dfA.columns.tolist())

Columns in A: ['CustomerId', 'Surname', 'CreditScore', 'Geography', 'Gender', 'Age', 'Tenure', 'Balance', 'NumOfProducts', 'HasCrCard', 'IsActiveMember', 'EstimatedSalary', 'Exited']


In [29]:

# 3. Calculate Average Credit Score
# Replace 'CreditScore' with the actual column name from the print output above
if 'CreditScore' in dfA.columns:
    avg_score = dfA['CreditScore'].mean()
    print(f"Average Credit Score: {avg_score}")

# 4. Count unique countries
# Replace 'Country' with the actual column name
if 'Country' in dfA.columns:
    num_countries = dfA['Country'].nunique()
    print(f"Number of different countries: {num_countries}")

Average Credit Score: 650.1172222222223
